library/hls/src/main/java/com/google/android/exoplayer2/source/hls/HlsSampleStreamWrapper.java

```diff
@@ -951,6 +1070,10 @@ private SampleQueue createSampleQueue(int id, int type) {
     sampleQueueTrackIds = Arrays.copyOf(sampleQueueTrackIds, trackCount + 1);
     sampleQueueTrackIds[trackCount] = id;
     sampleQueues = Util.nullSafeArrayAppend(sampleQueues, sampleQueue);
+    HlsMediaChunk mediaChunk = findChunkMatching(sourceChunk.uid);
```

Comment:
Again, I think the returned media chunk should be identical to sourceChunk, so no need to call this method. Also means that findChunkMatching can probably be removed.